src/cryptsetup/cryptsetup.c

```diff
@@ -819,6 +820,14 @@ static int measure_volume_key(
                 return 0;
         }
 
+        r = efi_stub_measured();
+        if (r < 0)
+                return log_debug_errno(r, "Failed to detect if we are running on a kernel image with TPM measurement enabled: %m");
+        if (r == 0) {
+                log_debug("Kernel stub did not measure kernel image, skipping userspace measurement, too.");
```

Comment:
Hmm, strictly speaking "did not measure" is quite different from "did not measure into the expected PCR".